src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -121,6 +121,160 @@ public Vector2 Center
             set => Translate(value - Center);
         }
 
+        // --
+
+        public float Width
+        {
+            get => _max.X - _min.X;
+            set => _max.X = _min.X + value;
+        }
+
+        public float Height
+        {
+            get => _max.Y - _min.Y;
+            set => _max.Y = _min.Y + value;
+        }
+
+        public float Left
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Top
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float Right
+        {
+            get => _max.X;
+            set => _max.X = value;
+        }
+
+        public float Bottom
+        {
+            get => _max.Y;
+            set => _max.Y = value;
+        }
+
+        public float X
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Y
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float SizeX
+        {
+            get => _max.X - _min.X;
+        }
+
+        public float SizeY
+        {
+            get => _max.Y - _min.Y;
+        }
+
+        public Vector2 Location => _min;
+
+        public bool IsEmpty => _min.X == 0 && _min.Y == 0 && _max.X == 0 && _max.Y == 0;
+
+        public static readonly Box2 Zero = new Box2(0, 0, 0, 0);
+
+        public static readonly Box2 Identity = new Box2(0, 0, 1, 1);
```

Comment:
`Identity` is probably not the best name here. My vote is for `UnitBox`, `UnitSquare` or `UnitRect` as that conveys better what it is.

Reply:
Good Point. I will name it `UnitSquare`

Reply:
tbh I would expect a `UnitSquare` to be centered at the origin, but I can't think of a better name either.

Reply:
This is not what is expected in math terms, there is the square from 0,0 to 1,1.
https://en.wikipedia.org/wiki/Unit_square

Reply:
yeah then it is the perfect name. Was just expecting it that way because of the unit circle/spheres being centered at the origin.